shared/instance.go

```diff
@@ -162,6 +162,20 @@ func IsRootDiskDevice(device map[string]string) bool {
 	return false
 }
 
+// IsNetworkMAC validates an ethernet MAC address. e.g. "32:47:ae:06:22:f9".
+func IsNetworkMAC(value string) error {
```

Comment:
Should we switch over to `net.ParseMAC` or similar functions?

Reply:
@stgraber yes we could do, originally this function enforced lower case only MACs, but now its been relaxed I see no reason we can't switch to use net.ParseMAC, will do.

Reply:
@stgraber have updated that MAC validation function in here https://github.com/lxc/lxd/pull/7693, I noticed that `net.ParseMAC` also validates infiniband addresses, so could also switch over that one, and because of that I've also added a length validation to our `IsNetworkMAC` to enforce only Ethernet MACs are valid.